File: frontend/src/types/product.ts
export type BundleSize = 1 | 2 | 3;

export interface Ingredient {
  name: string;
  role: string;
  country?: string;
}

export interface ProductVariant {
  sku: string;
  bundleSize: BundleSize;
  labelAr: string;
  priceKwd: number;
  compareAtPriceKwd?: number | null;
  isUpsellOffer?: boolean;
  stockQty?: number | null;
}

export interface ProductReviewBreakdown {
  rating: number;
  count: number;
}

export interface Product {
  slug: string;
  skuPrefix: "GUM" | "SHA" | "MAS";
  titleAr: string;
  titleEn: string;
  subtitleAr: string;
  shortDescriptionAr: string;
  originCountry: "US" | "JP" | "KR";
  originFlag: string;
  benefitsAr: readonly string[];
  ingredientsAr: readonly Ingredient[];
  mechanismAr: string;
  usageAr: string;
  whoIsForAr: readonly string[];
  whoIsNotForAr: readonly string[];
  faqs: ReadonlyArray<{ q: string; a: string }>;
  proofPoints: readonly string[];
  certifications: readonly string[];
  crossSellSlugs: readonly string[];
  upsellPartnerSlug: string;
  variants: readonly ProductVariant[];
  ratingValue: number;
  reviewCount: number;
  imageSlots: readonly string[];
  category: string;
}

File: backend/src/alnadara/schemas/product.py
from __future__ import annotations

from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class IngredientOut(BaseModel):
    name: str
    role: str
    country: str | None = None


class VariantOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    sku: str
    bundle_size: int
    label_ar: str
    price_kwd: Decimal
    compare_at_price_kwd: Decimal | None = None
    is_upsell_offer: bool = False
    stock_qty: int | None = None


class ProductOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    title_ar: str
    subtitle_ar: str | None = None
    short_description_ar: str | None = None
    origin_country: str | None = None
    ingredients: list[IngredientOut] = Field(default_factory=list)
    benefits: list[str] = Field(default_factory=list)
    usage_ar: str | None = None
    who_is_for_ar: str | None = None
    who_is_not_for_ar: str | None = None
    variants: list[VariantOut] = Field(default_factory=list)


class ProductsList(BaseModel):
    products: list[ProductOut]
